src/components/SearchFilter/SearchFilter.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState, useEffect } from 'react'
+import {
+  Grid,
+  Checkbox,
+  Dropdown,
+  Button,
+  Label,
+} from '@trussworks/react-uswds'
+import styles from './SearchFilter.module.scss'
+import { useSearchContext } from 'stores/searchContext'
+
+type PropTypes = {
+  labels: {
+    name: string
+  }[]
+}
+
+const SearchFilter = ({ labels }: PropTypes) => {
+  const { searchQuery } = useSearchContext()
+
+  const [searchPageFilters, setSearchPageFilters] = useState<string[]>([])
+
+  // Manages the state of the checkboxes and dropdown in the filter
+  const [filterItems, setFilterItems] = useState(
+    localStorage.getItem('filterItems')
+      ? JSON.parse(localStorage.getItem('filterItems') as string)
+      : {
+          application: false,
+          news: false,
+          documentation: false,
+          dropdown: '',
+        }
+  )
+
+  // When the filterItems state changes, update local storage
+  useEffect(() => {
+    localStorage.setItem('filterItems', JSON.stringify(filterItems))
+  }, [filterItems])
+
+  // Load the filterItems state from local storage
+  useEffect(() => {
+    const savedFilterItems = localStorage.getItem('filterItems')
+    if (savedFilterItems) {
+      setFilterItems(JSON.parse(savedFilterItems))
+
+      // If there is a true value in the filterItems state, add it to the searchPageFilters array with a category: prefix
+      const filterItemsArray = Object.entries(JSON.parse(savedFilterItems))
+      const filterItemsToSearch = filterItemsArray
+        .filter((item) => item[1] === true)
+        .map((item) => `category:${item[0]}`)
+      setSearchPageFilters(filterItemsToSearch)
+    }
+  }, [])
+
+  // This function updates the searchPageFilters array based on the checkboxes
+  const updateCheckedItems = (queryValue: string) => {
+    if (searchPageFilters.includes(queryValue)) {
+      const index = searchPageFilters.indexOf(queryValue)
+      const queryArray = [...searchPageFilters]
+      queryArray.splice(index, 1)
+      setSearchPageFilters(queryArray)
+    } else {
+      setSearchPageFilters([...searchPageFilters, queryValue])
+    }
+  }
```

Comment:
Yeah that is the expectation. I was thinking that if the user added some filters, searched, and then decided they wanted to update what is in the input in `<Search />` we need to make sure we tack that onto the end of the list of filters. I'm now wondering if it would be better to "clean up" the search query. Just show the user's query in the input instead of `category:application` etc.

Reply:
Wait. I actually just gave the parameter a terrible name. `queryValue` is the the checkbox that has just been checked/unchecked. I initially wanted this function to be generalized and be able to update the array of search filters for checkboxes, the dropdown, whatever else. That being said, I'm updating the param name to be something more intuitive. To clarify your original question, this function just looks to see if the most recently checked item already exists in the `searchPageFilters` array, and if it doesn't, it is tacked onto the end of the array.

Reply:
Also, I'm addressing the issue you mentioned by moving the `searchPageFilters` to `SearchContext` and changing the way that the search form is submitted to accommodate both options: clicking the `Filter` button or the `Search` button will result in applying any selected filters